src/util/expr_util.cpp

```diff
@@ -71,6 +71,13 @@ exprt gen_zero(const typet &type, bool array_as_array_of)
         result.copy_to_operands(gen_zero(type.subtype(), array_as_array_of));
     }
   }
+  else if(type_id == "vector")
```

Comment:
Shall we create a proper ID in our irep for `vector`?

Reply:
Yes, it was created at irep